client/my-sites/people/edit-team-member-form/edit-user-form.jsx

```diff
@@ -40,30 +40,32 @@ const debug = debugModule( 'calypso:my-sites:people:edit-team-member-form' );
 class EditUserForm extends Component {
 	state = this.getStateObject( this.props );
 
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		this.setState( this.getStateObject( nextProps ) );
-	}
-
-	getRole( roles ) {
-		return roles && roles[ 0 ] ? roles[ 0 ] : null;
+	componentDidUpdate() {
+		if ( ! this.hasUnsavedSettings() ) {
+			this.props.markSaved();
+		}
 	}
 
 	getStateObject( props ) {
```

Comment:
Or we could even destructure `props` here:

```suggestion
	getStateObject( { user, isExternalContributor } ) {
```

and that way we won't need to access the entire `props` object below.

Reply:
Good point 👍🏻  added via https://github.com/Automattic/wp-calypso/pull/51278/commits/ef56f10c03c2426457f6cbbc1625aeab3ab7f5af